contracts/interchain-token-service/InterchainTokenService.sol

```diff
@@ -878,7 +964,7 @@ contract InterchainTokenService is
             // slither-disable-next-line reentrancy-events
             emit TokenSent(tokenId, destinationChain, destinationAddress, amount);
 
-            payload = abi.encode(SELECTOR_RECEIVE_TOKEN, tokenId, destinationAddress, amount);
+            payload = abi.encode(SELECTOR_RECEIVE_TOKEN, tokenId, sourceAddress.toBytes(), destinationAddress, amount);
```

Comment:
Why are you adding the source address here? It seems it is only to emit an event when receiving, and I'd rather save the non-trivial amount of gas instead.

Reply:
I mean without it on the destination chain you don't even know who sent the token? Token transfers shouldn't be anon right?